hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java

```diff
@@ -0,0 +1,142 @@
+            snapshot1, snapshot2, null, 0, false);
+
+    assertEquals(IN_PROGRESS, response.getJobStatus());
+
+    String oldLeader = getCluster().getOMLeader().getOMNodeId();
+
+    OzoneManager omLeader = getCluster().getOMLeader();
+    getCluster().shutdownOzoneManager(omLeader);
+    getCluster().restartOzoneManager(omLeader, true);
+
+    await().atMost(Duration.ofSeconds(120))
+        .until(() -> getCluster().getOMLeader() != null);
+
+    String newLeader = getCluster().getOMLeader().getOMNodeId();
+
+    if (Objects.equals(oldLeader, newLeader)) {
+      // If old leader becomes leader again. Job should be done by this time.
+      response = getObjectStore().snapshotDiff(volumeName, bucketName,
+          snapshot1, snapshot2, null, 0, false);
+      assertEquals(DONE, response.getJobStatus());
+      assertEquals(100, response.getSnapshotDiffReport().getDiffList().size());
+    } else {
+      // If new leader is different from old leader. SnapDiff request will be
+      // new to OM, and job status should be IN_PROGRESS.
+      response = getObjectStore().snapshotDiff(volumeName, bucketName,
+          snapshot1, snapshot2, null, 0, false);
+      assertEquals(IN_PROGRESS, response.getJobStatus());
+      while (true) {
+        response =
+            getObjectStore().snapshotDiff(volumeName, bucketName, snapshot1,
+                snapshot2, null, 0, false);
+        if (DONE == response.getJobStatus()) {
+          assertEquals(100,
+              response.getSnapshotDiffReport().getDiffList().size());
+          break;
+        }
+        Thread.sleep(response.getWaitTimeInMs());
+      }
+    }
+  }
+
+  @Test
+  public void testSnapshotIdConsistency() throws Exception {
+    OzoneBucket ozoneBucket = setupBucket();
+    String volumeName = ozoneBucket.getVolumeName();
+    String bucketName = ozoneBucket.getName();
+
+    createKey(ozoneBucket);
+
+    String snapshotName = "snap-" + RandomStringUtils.randomNumeric(5);
+
+    getObjectStore().createSnapshot(volumeName, bucketName, snapshotName);
+    List<OzoneManager> ozoneManagers = getCluster().getOzoneManagersList();
+    List<String> snapshotIds = new ArrayList<>();
+
+    for (OzoneManager ozoneManager : ozoneManagers) {
+      await().atMost(Duration.ofSeconds(120))
+          .until(() -> {
+            SnapshotInfo snapshotInfo;
+            try {
+              snapshotInfo = ozoneManager.getMetadataManager()
+                  .getSnapshotInfoTable()
+                  .get(SnapshotInfo.getTableKey(volumeName,
+                      bucketName,
+                      snapshotName));
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+
+            if (snapshotInfo != null) {
+              snapshotIds.add(snapshotInfo.getSnapshotID());
+            }
+            return snapshotInfo != null;
+          });
+    }
+
+    assertEquals(1, snapshotIds.stream().distinct().count());
```

Comment:
Is snapshotId assertion enough? I guess we should also check rocks db contents.

Reply:
Checking only `snapshotId` should be enough.

Reply:
the list of sst files should be same I believe.

Reply:
I don't think that should be tested as part of this test.